src/nvim/highlight_group.c

```diff
@@ -1568,7 +1568,16 @@ Dictionary ns_get_hl_defs(NS ns_id, Dict(get_highlight) *opts, Arena *arena, Err
   Boolean link = GET_BOOL_OR_TRUE(opts, get_highlight, link);
   int id = -1;
   if (HAS_KEY(opts, get_highlight, name)) {
-    id = syn_check_group(opts->name.data, opts->name.size);
+    Boolean create = false;
+    if (HAS_KEY(opts, get_highlight, create)) {
+      create = opts->create;
+    }
+    id = create ? syn_check_group(opts->name.data, opts->name.size)
+                : syn_name2id_len(opts->name.data, opts->name.size);
+    if (id == 0 && strchr(opts->name.data, ' ') == NULL) {
```

Comment:
This is a bit specific, `' '` is not the only invalid char. could check `id =≠ 0 && !create` instead.